app/src/main/java/com/android/template/utils/auth/AuthUtils.kt

```diff
@@ -0,0 +1,13 @@
+package com.android.template.utils.auth
+
+import android.content.Intent
+
+interface AuthUtils {
+    fun getRequestCode(): Int
+
+    // TODO: maybe add overloads for theme, logo, etc. or:
+    // TODO: maybe add private variables of logo value, theme value, etc. and set setters
+    fun buildSignInIntent(): Intent
```

Comment:
2/3 metode u ovoj klasi generiraju nešto, tako da je ova klasa potencijalno `Factory`

Reply:
`AuthenticationIntentFactory`